openshift-knative-operator/pkg/common/deployment.go

```diff
@@ -0,0 +1,73 @@
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// InjectEnvironmentIntoDeployment injects the specified environment variables into the
+// specified deployment/container.
+// Note: We're not deleting empty environment variables and instead set them to empty
+// string. Three-way-merging of the deployment drops the update otherwise.
+func InjectEnvironmentIntoDeployment(deploymentName, containerName string, env map[string]string) mf.Transformer {
+	return transformDeployment(deploymentName, func(deploy *appsv1.Deployment) error {
+		containers := deploy.Spec.Template.Spec.Containers
+		for i := range containers {
+			c := &containers[i]
+			if c.Name != containerName {
+				continue
+			}
+
+			for key, value := range env {
+				c.Env = upsert(c.Env, key, value)
```

Comment:
Do we add this to the rbac proxy container too?

Reply:
Currently, we don't. This specifically targets the controller container to enable digest resolution to work properly even in a HTTP_PROXY settings. AFAIK we don't have other components that might call to the outside and thus need those settings.

Reply:
Ok I see that we skip if the container name does not match good.